### PR TITLE
CMR-7079 function for retrieving s3 bucket information in access-control

### PR DESCRIPTION
**Pull request by jaybarra**

### CODE REVIEW

**Review by jceaser — APPROVED** [review]

File: access-control-app/src/cmr/access_control/services/acl_util.clj [hunk 1]
```diff
@@ -150,3 +151,29 @@
...
+  (let [metadata (umm-spec/parse-metadata (assoc context :ignore-kms-keywords true) concept)
+        s3-data (get-in metadata [:DirectDistributionInformation
+                                  :S3BucketAndObjectPrefixNames])]
+    {(:concept-id concept) s3-data}))
+
+(defn s3-bucket-and-prefixes-for-collection-ids
+  "Takes a list of collection IDs and returns a map of collection-id and
+   s3-bucket-and-object-prefix-names values."
+  [context coll-ids]
+  (let [concepts (pmap #(mdb/get-latest-concept context %) coll-ids)]
```
Comment (jaybarra) [inline access-control-app/src/cmr/access_control/services/acl_util.clj:168]: consider timing this call

Comment (jaybarra) [reply access-control-app/src/cmr/access_control/services/acl_util.clj:168]: add warning log for large (> 2000?) number of ids passed in

File: access-control-app/src/cmr/access_control/services/acl_util.clj [hunk 2]
```diff
@@ -150,3 +151,26 @@
...
+(defn-timed s3-bucket-and-prefixes-for-collection-ids
+  "Takes a list of collection IDs and returns a map of collection-id and
+   s3-bucket-and-object-prefix-names values."
+  [context coll-ids]
+  (when (> (count coll-ids) max-s3-bucket-query-size)
+    (warn (format (str "%d concept IDs were submitted to query mdb. "
+                       "Consider breaking this call into smaller batches.")
+                  (count coll-ids))))
+  (let [concepts (pmap #(mdb/get-latest-concept context %) coll-ids)]
+    (into {} (map #(s3-bucket-and-prefix-names-concept-id-map context %)
```
Comment (mschmele) [inline access-control-app/src/cmr/access_control/services/acl_util.clj:175]: Do we want to remove any items in this map where the s3 urls may be nil?

File: access-control-app/int-test/cmr/access_control/int_test/acl_util.clj [hunk 3]
```diff
@@ -34,3 +36,32 @@
             {:group-permissions [{:user-type "guest", :permissions ["create" "delete"]}], :system-identity {:target "TAG_GROUP"}}
             :delete
             "User: [admin] Deleted ACL [{:group-permissions [{:user-type \"guest\", :permissions [\"create\" \"delete\"]}], :system-identity {:target \"TAG_GROUP\"}}]"))))
+
+
+(deftest s3-bucket-and-prefixes-for-collection-ids-test
```
Comment (mschmele) [inline access-control-app/int-test/cmr/access_control/int_test/acl_util.clj:41]: Add tests for edge cases like nil or malformed s3 urls if the schema will allow

**Comment by jaybarra** [comment]
Rejecting as standalone functionality. It will be integrated into work for CMR-7080 where a superior alternative has been proposed.
